src/internal/partial_solution.rs

```diff
@@ -80,8 +80,7 @@ impl<P: Package, V: Version> PartialSolution<P, V> {
             .rposition(|(l, _)| *l == decision_level)
             .unwrap_or(self.history.len() - 1);
         *self = Self::from_assignments(
-            self.history
-                .to_owned()
+            std::mem::take(&mut self.history)
```

Comment:
This must be working as intended since tests pass, but I would like to ask for my own understanding.
`std::mem::take` replaces `self.history` with a default value, i.e. empty `Vec` in this case. Before the change the history was retained.

Why doing this is correct in this case?

Reply:
Because we are building a new self and its history and assigning it to `*self`.

old code:
```rust
let copy = self.history.to_owned();
let new_self = Self::from_assignments(copy);
*self = new_self;
```

Note how no one uses the `self.history` after the copy. It is just replaced as part of the last line.

Reply:
Didn't see that, thank you for the explanation!